### ISSUE TITLE
Add a new canRated flag to custom starting positions 

### ISSUE THREAD

**Issue by gbtami** [gbtami/pychess-variants#1510]

**Comment by PraseodymiumSpike** [comment]
I disagree with rating custom starting positions as it is meant to be a rating in the original variant. Chess ratings should reflect rating in chess, not in any variant, even if that variant has a relatively minor change like removing castling.

**Comment by gbtami** [comment]
Original idea came from ubdip on Discord https://discord.com/channels/634298688663191582/634298688663191584/1251122012777218069

"One reasonable suggestion to get more custom starting positions rated I could see is to have a flag on each predefined custom starting positions that says whether it would still get rated for its variant (but not separately). E.g., for embassy, gothic, etc. it could be rated for capablanca, but upside-down won't get rated for chess. Similarly no-castling could get rated for chess. Just to get more data for reliable ratings in the original variant itself."

"There are two differences in custom starting positions. They can be very different dynamically (e.g., upside-down vs chess) and therefore not represent normal playing strength, and even more importantly they can be unbalanced. E.g., shogi handicap positions also are "custom starting positions", so rating them without adjustment would be meaningless, and very likely some people would exploit it. So only positions that are balanced and similar to the normal starting position can be rated."

**Comment by gbtami** [comment]
Implemented via https://github.com/gbtami/pychess-variants/commit/faa3c99751de19770f4461bf398a42a5e64abb5e
